analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1288 @@
+
+        var argument = ArgumentDescriptor.AttributeProperty("AttributeUsage", propertyName);
+        var (result, context) = MatchArgumentCS(model, node, argument);
+        result.Should().Be(expected);
+        if (result)
+        {
+            // The mapped parameter is the "value" parameter of the property set method.
+            context.Parameter.Name.Should().Be("value");
+            var method = context.Parameter.ContainingSymbol.Should().BeAssignableTo<IMethodSymbol>().Which;
+            method.MethodKind.Should().Be(MethodKind.PropertySet);
+            method.AssociatedSymbol.Should().BeAssignableTo<IPropertySymbol>().Which.Name.Should().Be(propertyName);
+        }
+    }
+
+    [DataTestMethod]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true)>""", "AllowMultiple", true)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true, Inherited := true)>""", "AllowMultiple", true)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  $$AllowMultiple := true, Inherited := true)>""", "Inherited", false)]
+    [DataRow("""<AttributeUsage(AttributeTargets.All,  AllowMultiple := true, $$Inherited := true)>""", "Inherited", true)]
+    public void Attribute_Property_VB(string attribute, string propertyName, bool expected)
+    {
+        var snippet = $$"""
+            Imports System
+
+            {{attribute}}
+            Public NotInheritable Class TestAttribute
+                Inherits Attribute
+            End Class
+            """;
+        var (node, model) = ArgumentAndModelVB(snippet);
+
+        var argument = ArgumentDescriptor.AttributeProperty("AttributeUsage", propertyName);
+        var (result, context) = MatchArgumentVB(model, node, argument);
+        result.Should().Be(expected);
+        if (result)
+        {
+            // The mapped parameter is the "value" parameter of the property set method.
+            context.Parameter.Name.Should().Be("value");
+            var method = context.Parameter.ContainingSymbol.Should().BeAssignableTo<IMethodSymbol>().Which;
+            method.MethodKind.Should().Be(MethodKind.PropertySet);
+            method.AssociatedSymbol.Should().BeAssignableTo<IPropertySymbol>().Which.Name.Should().Be(propertyName);
+        }
+    }
+
+    private static string WrapInMethodCS(string snippet) =>
+        $$"""
+        using System;
+        class C
+        {
+            public void M()
+            {
+                {{snippet}}
+            }
+        }
+        """;
+
+    private static string WrapInMethodVB(string snippet) =>
+        $$"""
+        Imports System
+        Class C
+            Public Sub M()
+                {{snippet}}
+            End Sub
+        End Class
+        """;
+
+    private static (SyntaxNode Node, SemanticModel Model) ArgumentAndModel(string snippet,
+        Func<string, MetadataReference[], (SyntaxTree Tree, SemanticModel Model)> compile, params Type[] argumentNodeTypes)
```

Comment:
You can also remove this as a parameter and have a predefined list that includes all the types. It will still work since the syntax tree for c# will not contain vb types. And the other way around. Up to you.

Reply:
I'd like to keep it.